pkg/traceql/test_examples.yaml

```diff
@@ -0,0 +1,229 @@
+# valid queries parse successfully and return nil when calling .validate()
+valid:
+  # spanset filters
+  - '{ true }'
+  - '{ !true }'
+  - '{ true && false }'
+  - '{ true || false }'
+  - '{ 1 = 2 }'
+  - '{ 1 != 2 }'
+  - '{ 1 > 2 }'
+  - '{ 1 >= 2 }'
+  - '{ 1 < 2 }'
+  - '{ 1 <= 2 }'
+  - '{ 1 + 1 = 2 }'
+  - '{ 1 - 1 = 2 }'
+  - '{ 1 * 1 = 2 }'
+  - '{ 1 / 1 = 2 }'
+  - '{ 1 ^ 1 = 2 }'
+  - '{ -1 = 2 }'
+  - '{ "test" =~ "test" }'
+  - '{ "test" !~ "test" }'
+  - '{ "test" = "test" }'
+  - '{ "test" != "test" }'
+  - '{ .a }'
+  - '{ !.a }'
+  - '{ .a && false }'
+  - '{ .a || true }'
+  - '{ .a = 2 }'
+  - '{ .a != 2 }'
+  - '{ .a > 2 }'
+  - '{ .a >= 2 }'
+  - '{ .a < 2 }'
+  - '{ .a <= 2 }'
+  - '{ .a + 1 = 2 }'
+  - '{ .a - 1 = 2 }'
+  - '{ .a * 1 = 2 }'
+  - '{ .a / 1 = 2 }'
+  - '{ .a ^ 1 = 2 }'
+  - '{ -.a = 2 }'
+  - '{ .a =~ "test" }'
+  - '{ .a !~ "test" }'
+  - '{ .a = "test" }'
+  - '{ .a != "test" }'
+  - '{ parent.a != 3 }'
+  - '{ parent.resource.a && true }'
+  - '{ parent.span.a > 3 }'
+  - '{ parent.duration = 1h }'
+  - '{ resource.a != 3 }'
+  - '{ span.a != 3 }'
+  - '{ !("test" != .c || ((true && .b) || 3 < .a)) }'
+  - '{ parent = nil }'
+  - '{ status = ok }'
+  - '{ status = unset }'
+  - '{ status = error }'
+  - '{ status != error }'
+  - '{ duration > 1s }'
+  - '{ duration > 1s * 2s }' 
+  - '{ .foo = nil }'
+  - '{ 1 = childCount }'
+  - '{ 1 * 1h = 1 }'     # combining float, int and duration can make sense, but can also be weird. we just accept it all
+  - '{ 1 / 1.1 = 1 }'
+  - '{ 1 < 1h }'
+  - '{ 1 <= 1.1 }'
+  # spanset expressions
+  - '{ true } && { true }'
+  - '{ true } || { true }'
+  - '{ true } >> { true }'
+  - '{ true } > { true }'
+  - '{ true } ~ { true }'
+  # scalar filters
+  - 'avg(.field) > 1'
+  - 'sum(3) = 2'
+  - 'min(childCount) < 2'
+  - 'max(duration) >= 1s'
+  - 'avg(.field) > avg(1)'
```

Comment:
I think I understand why `aggregate(static)` is valid (fieldExpression definition includes all attribute lookups/intrinsics/status).  It doesn't seem useful and would prefer if it weren't parseable, and probably not worth changing the grammar for.  But wondering if these statements should be included in the "valid" list?

Reply:
It would be difficult to remove from the grammar, but not particularly difficult to add to the validate logic. We could descend the field expression and make sure that at least one element was an attribute or intrinsic. 

Will make this change.